src/lsp/LSPInterface.js

```diff
@@ -0,0 +1,150 @@
+
+/*
+ * Copyright (c) 2018 - present Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+ define(function (require, exports, module) {
+    "use strict";
+    var NodeDomain              = require("utils/NodeDomain"),
+    AppInit                     = require("utils/AppInit"),
+    ExtensionUtils              = require("utils/ExtensionUtils"),
+    FileUtils                   = require("file/FileUtils");
+
+    var lspServerList = [];
+    var serverDomainList = {};
+    var $deferredResponse = {};
+    var _domainName = "lspServers";
+    var _serverDomain = null;
+    var _bracketsPath   = FileUtils.getNativeBracketsDirectoryPath(),
+        _modulePath = FileUtils.getNativeModuleDirectoryPath(module),
+        _nodePath = "node/LSPServerDomain",
+        _domainPath = [_bracketsPath, _modulePath, _nodePath].join("/");
+    var _callbackList = {};
+    var _requestId = {};
+
+    /**
+     * Get Unique request id for the request being made to LSP Server
+     * @param   {String} serverName - server where message is to be sent
+     * @returns {Number} requestId - requestId to be associated with the request
+     */
+    function getNextReqId(serverName){
+        if(!(serverName in _requestId)){
+            _requestId[serverName] = 1;
+            return 1;
+        }
+        return ++_requestId[serverName];
+    }
+
+    /**
+     * Receive any response/notification from the LSP server
+     * @param {Object}  event  - event received from node
+     * @param {Object}  msgObj - json containing - {
+     *                          serverName - name of the LSP server which sent the message
+     *                          param - Content of the message
+     */
+    function receiveMessageFromServer(event, msgObj) {
+        
+        if(msgObj.param.method && _callbackList[msgObj.serverName+msgObj.param.method]){
```

Comment:
I would assert that `msgObj` is defined and param exists before checkin for `method`. Same applies for accessing `_callbackList`. There is a some potential to create a variable and assign the constructed key `msgObj.serverName+msgObj.param`.